lib/js/src/manager/screen/choiceset/_ChoiceSetManagerBase.js

```diff
@@ -131,62 +124,39 @@ class _ChoiceSetManagerBase extends _SubManagerBase {
      * @param {ChoiceCell[]} choices - A list of ChoiceCell objects that will be part of a choice set later
-     * @returns {Promise} - A promise that resolves to a Boolean of whether the operation is a success
+     * @returns {Promise} - A promise that resolves to a Boolean of whether the operation is added successfully
      */
-    preloadChoices (choices = null) {
+    preloadChoices (choices = []) {
         if (this._getState() === _SubManagerBase.ERROR) {
             console.warn('ChoiceSetManager: Choice Manager In Error State');
             return Promise.resolve(false);
         }
-        const choicesToUpload = this._getChoicesToBeUploadedWithArray(choices);
-        this._removeChoicesFromChoices(this._preloadedChoices, choicesToUpload);
-        this._removeChoicesFromChoices(this._pendingPreloadChoices, choicesToUpload);
-        if (choicesToUpload.length === 0) {
+        if (choices.length === 0) {
             return Promise.resolve(true);
         }
-        this._updateIdsOnChoices(choicesToUpload);
+        this._updateIdsOnChoices(choices);
         // Add the preload cells to the pending preload choices
-        this._pendingPreloadChoices = this._pendingPreloadChoices.concat(choicesToUpload);
         if (this._fileManager === null) {
             console.error('ChoiceSetManager: File Manager was null in preload choice operation');
             return Promise.resolve(false);
         }
+
         return new Promise(resolve => {
-            const preloadChoicesOperation = new _PreloadChoicesOperation(this._lifecycleManager, this._fileManager,
-                this._displayName, this._defaultMainWindowCapability, this._isVrOptional, choicesToUpload, (success, failedChoiceCells) => {
-                    if (success) {
-                        this._preloadedChoices = this._preloadedChoices.concat(choicesToUpload);
-                        this._removeChoicesFromChoices(choicesToUpload, this._pendingPreloadChoices);
-                    } else {
-                        this._removeChoicesFromChoices(failedChoiceCells, choicesToUpload);
-                        this._preloadedChoices = this._preloadedChoices.concat(choicesToUpload);
-                        this._removeChoicesFromChoices(choicesToUpload, this._pendingPreloadChoices);
+            const preloadChoicesOperation = new _PreloadPresentChoicesOperation(this._lifecycleManager, this._fileManager, this._displayName,
+                this._defaultMainWindowCapability, this._isVrOptional, choices,
+                this._preloadedChoices, null, null, null, null, null, null, (loadedCells, success) => {
+                    if (!success) {
+                        console.error('ChoiceSetManager: There was an error preloading choice cells');
+                        return resolve(false);
                     }
-                    resolve(success);
+                    this._preloadedChoices = loadedCells.map(cell => cell.clone());
```

Comment:
The cloning does need to happen in the other operations, however doing this at the end of the operation caused issues since an operation could be started and the cells could be updated while the task is running (in some cases this would happen after the cells had been preloaded but before being passed back to the manager). I've moved the cloning to happen during the `_updateCells...` methods which are called at the start of the operations.